reverse_search.go

```diff
@@ -0,0 +1,54 @@
+package prompt
+
+import (
+	"strings"
+)
+
+const (
+	matchSearchPrefixFmt = "(reverse-i-search)`%s':"
+	failSearchPrefixFmt  = "(failed reverse-i-search)`%s':"
+)
+
+// reverseSearchInfo contains info about reverseSearch state.
+type reverseSearchState struct {
+	// history, with which reverse-search works.
+	history *History
+	// historyPrefix is the last index, included to the search scope.
```

Comment:
It is unclear from the var name that this is an index. Index of what?

Reply:
Renamed.

Reply:
Just to get a better understanding, is this an index the search will start from on next reverse search?